homeassistant/components/lidarr/config_flow.py

```diff
@@ -0,0 +1,156 @@
+                if isinstance(result, tuple):
+                    user_input[CONF_API_KEY] = result[1]
+                elif isinstance(result, str):
+                    errors = {"base": result}
+            except exceptions.ArrAuthenticationException:
+                errors = {"base": "invalid_auth"}
+            except (ClientConnectorError, exceptions.ArrConnectionException):
+                errors = {"base": "cannot_connect"}
+            except exceptions.ArrException:
+                errors = {"base": "unknown"}
+            if not errors:
+                if self.entry:
+                    self.hass.config_entries.async_update_entry(
+                        self.entry, data=user_input
+                    )
+                    await self.hass.config_entries.async_reload(self.entry.entry_id)
+
+                    return self.async_abort(reason="reauth_successful")
+
+                return self.async_create_entry(
+                    title=DEFAULT_NAME,
+                    data=user_input,
+                    options={
+                        CONF_MAX_RECORDS: DEFAULT_MAX_RECORDS,
+                    },
+                )
+
+        user_input = user_input or {}
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_URL, default=user_input.get(CONF_URL, DEFAULT_URL)
+                    ): str,
+                    vol.Optional(CONF_API_KEY): str,
+                    vol.Optional(
+                        CONF_VERIFY_SSL,
+                        default=user_input.get(CONF_VERIFY_SSL, False),
+                    ): bool,
+                }
+            ),
+            errors=errors,
+        )
+
+
+async def validate_input(
+    hass: HomeAssistant, data: dict[str, Any]
+) -> tuple[str, str, str] | str | SystemStatus:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    lidarr = LidarrClient(
+        api_token=data.get(CONF_API_KEY, ""),
+        url=data[CONF_URL],
+        session=async_get_clientsession(hass),
+        verify_ssl=data[CONF_VERIFY_SSL],
+    )
+    if CONF_API_KEY not in data:
+        return await lidarr.async_try_zeroconf()
+    return await lidarr.async_get_system_status()
+
+
+class LidarrOptionsFlowHandler(OptionsFlow):
+    """Handle Lidarr client options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, int] | None = None
+    ) -> FlowResult:
+        """Manage Lidarr options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_MAX_RECORDS,
```

Comment:
Why is this configurable ?

Reply:
This is for queue and wanted items. Setting the default down to 50. Are you saying we should not offer this option because it would otherwise allow users to unwittingly bloat their database?

Reply:
I think the idea is to have as few configuration options as possible.